### PR TITLE
Add kbdfans kbd67 mkii rgb v3

### PR DESCRIPTION
**Pull request by datafx**
KBDFans released a new version of their kbd67 mkii rgb pcb.  The new board uses a different rgbmatrix driver so the entire setup for v1 and v2 need to be changed to allow this since all the versions could no longer share code.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by datafx** [comment]
This board was designed by @moyi4681 so I don't know if he has a better implementation or suggestions for this PR.  I'm not sure if I should change the product id back to the default 0x0100 since that is what is in kbdfans VIA json they have on their website.  The rest of the boards were just incremented by one so that's what I did here.

File: keyboards/kbdfans/kbd67/mkiirgb/v1/v1.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#include "v1.h"
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v1/v1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd67/mkiirgb/v1/v1.c [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+{
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void)
+{
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v1/v1.c:109]: This is no longer needed, at all. 
```suggestion
```

File: keyboards/kbdfans/kbd67/mkiirgb/v1/v1.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#pragma once
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v1/v1.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd67/mkiirgb/v2/v2.c [hunk 4]
```diff
@@ -0,0 +1,119 @@
+#include "v2.h"
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v2/v2.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd67/mkiirgb/v2/v2.c [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+{
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void)
+{
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v2/v2.c:109]:
```suggestion
```

File: keyboards/kbdfans/kbd67/mkiirgb/v2/v2.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#pragma once
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v2/v2.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd67/mkiirgb/v3/config.h [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v3/config.h:45]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd67/mkiirgb/mkiirgb.h [hunk 8]
```diff
@@ -1,16 +1,28 @@
...
+
+#ifdef KEYBOARD_kbdfans_kbd67_mkiirgb_v1
+    #include "v1.h"
+#endif
+#ifdef KEYBOARD_kbdfans_kbd67_mkiirgb_v2
+    #include "v2.h"
+#endif
+#ifdef KEYBOARD_kbdfans_kbd67_mkiirgb_v3
+    #include "v3.h"
+#endif
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/mkiirgb.h:27]:
```suggestion
#if defined(KEYBOARD_kbdfans_kbd67_mkiirgb_v1)
    #include "v1.h"
#elif defined(KEYBOARD_kbdfans_kbd67_mkiirgb_v2
    #include "v2.h"
#elif defined(KEYBOARD_kbdfans_kbd67_mkiirgb_v3)
    #include "v3.h"
#endif
```

File: keyboards/kbdfans/kbd67/mkiirgb/v2/rules.mk [hunk 9]
```diff
@@ -28,6 +28,7 @@ MIDI_ENABLE = no            # MIDI support
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/kbdfans/kbd67/mkiirgb/v2/rules.mk:31]: Please remove this.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd67/mkiirgb/mkiirgb.h [hunk 10]
```diff
@@ -1,16 +1,26 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #pragma once
-#define XXX KC_NO
+
+#if defined(KEYBOARD_kbdfans_kbd67_mkiirgb_v1)
+    #include "v1.h"
+#elif defined(KEYBOARD_kbdfans_kbd67_mkiirgb_v2
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkiirgb/mkiirgb.h:21]:
```suggestion
#elif defined(KEYBOARD_kbdfans_kbd67_mkiirgb_v2)
```

File: keyboards/kbdfans/kbd67/mkiirgb/v1/rules.mk [hunk 11]
```diff
@@ -16,4 +16,4 @@ AUDIO_ENABLE = no
 RGB_MATRIX_ENABLE = yes            # Use RGB matrix
 RGB_MATRIX_DRIVER = IS31FL3731
 
-LAYOUTS = 65_ansi_blocker
+LAYOUTS = 65_ansi_blocker
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkiirgb/v1/rules.mk:19]:
```suggestion
LAYOUTS = 65_ansi_blocker

```

File: keyboards/kbdfans/kbd67/mkiirgb/v1/v1.c [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    1, 1, 1, 4, 1, 1, 1, 1, 1
+} };
+
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void)
+{
+    if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK))
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkiirgb/v1/v1.c:119]:
```suggestion
    if (host_keyboard_led_state().caps_lock)
```

File: keyboards/kbdfans/kbd67/mkiirgb/v2/v2.c [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+    1, 1, 1, 4, 1, 1, 1, 1, 1
+} };
+
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void)
+{
+    if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK))
```
Comment (fauxpark) [inline keyboards/kbdfans/kbd67/mkiirgb/v2/v2.c:119]:
```suggestion
    if (host_keyboard_led_state().caps_lock)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**